java-frontend/src/main/java/org/sonar/java/resolve/TypeInferenceSolver.java

```diff
@@ -71,13 +78,22 @@ TypeSubstitution inferTypeSubstitution(MethodJavaSymbol method, List<JavaType> f
 
   private TypeSubstitution inferTypeSubstitution(MethodJavaSymbol method, TypeSubstitution substitution, JavaType formalType, JavaType argumentType,
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "inferTypeSubstitution" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)